### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by c00p75**
# Blog app - Add forms

## Objectives

- [ ] Created method called current_user in ApplicationController to make current user data available to all controllers.

- [ ] Created form to create a post on behalf of the current_user.

- [ ] Created form to create a comment on behalf of the current_user.
- [ ] Created form to allow users to add likes to posts.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,9 @@
 Rails.application.routes.draw do
   root 'users#index'
   resources :users, only: [:index, :show] do
-    resources :posts, only: [:index, :show]
+    resources :posts do 
+      resources :comments, only: [:create]
```
Comment (rustam-tolipov) [inline config/routes.rb]: **You did an awesome job!**
We have two more requirements to complete. Let's get started. 

- [x] For starting, your comments should reponse to these routes: In your project I can only see posts#create, please posts#new route as well. Thank you!
```ruby
GET   /posts/new                   posts#new
  POST   /posts(.:format)                     posts#create
```

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class CommentsController < ApplicationController
+  def create
```
Comment (rustam-tolipov) [inline app/controllers/comments_controller.rb:8]: - [ ] In your comments controller there should be method called `new` as you have created in posts controller. Kindly add and resubmit it again. After completing these requirements please make sure it's working as expected.

**Review by rustam-tolipov — CHANGES_REQUESTED** [review]
Hello @c00p75,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
_Your project is almost complete!_
_No Linters errors_
_Well-implemented code_

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Check the comments under the review._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: config/routes.rb (hunk 1 above)
Comment (c00p75) [reply config/routes.rb]: Hello @Rustamxon7. Thanks for the prompt feedback. I'm unclear about this requirement. What I understood from the [docs](https://guides.rubyonrails.org/routing.html#crud-verbs-and-actions) is that ```resources :posts do```  creates all routes, including post#create and post#new. Are you referring to the comments route (``` resources :comments, only: [:create]```) instead?

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (c00p75) [reply app/controllers/comments_controller.rb:8]: Well noted. Thank you🙏

File: config/routes.rb (hunk 1 above)
Comment (rustam-tolipov) [reply config/routes.rb]: No, your posts have all routes that we need, what we need not in your comments add `:new` route in comment's only: [...] part. You can check by `rails routes` to see all routes you have.

Comment (c00p75) [reply config/routes.rb]: Understood. Thank you.

**Review by hiromataba — APPROVED** [review]
# Status: Approved :tada: :tada: :tada: 
Hi @c00p75     :wave: :wave: 

200w  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**